firmware/Core/Src/unit_tests/test_string_helpers.c

```diff
@@ -0,0 +1,97 @@
+#include "transforms/string_helpers.h"
+#include "unit_tests/unit_test_helpers.h"
+
+#include <stdint.h>
+#include <string.h>
+
+// Function which tests the split_string_by_delimiter function
+uint8_t TEST_EXEC__HELPER_split_string_by_delimiter()
```

Comment:
These look like some good tests. I'll propose maybe splitting these changes out into a separate branch in case anyone ever wants to use them. I'd prefer not to merge them until we have a use for them though